library/src/main/java/com/android/mms/transaction/DownloadManager.java

```diff
@@ -122,11 +129,29 @@ public static void finishDownload(String location) {
         }
     }
 
-    private static boolean isNotificationExist(Context context, String location) {
-        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.KITKAT) {
-            return true;
+    /**
+     * �_�E�����[�h�̏I���������s����
+     * @param context �R���e�L�X�g
+     * @param location �_�E�����[�h URL
+     */
+    public static void finishDownload(Context context, String location) {
+        finishDownload(location);
+        downloadIfDelayed(context);
+    }
+
+    /**
+     * �ڑ������߂̂��ߒx���������_�E�����[�h������ꍇ�͍Ď�s����
+     * @param context �R���e�L�X�g
+     */
+    private static void downloadIfDelayed(Context context) {
+        if (sDelayedDownloadInfoList.size() > 0) {
+            DownloadInfo downloadInfo = sDelayedDownloadInfoList.get(0);
+            new DownloadMultimediaMessageTask(context, downloadInfo).execute();
```

Comment:
If not specify the executor, AsyncTask will be executed in the common executor in application.
Please execute it in unique executor in order not to affect the application.

See: https://github.com/aTakayukiMorita/android-smsmms/pull/21